exchanges/okx/okx_test.go

```diff
@@ -0,0 +1,2841 @@
+
+func TestTransactionHistory(t *testing.T) {
+	t.Parallel()
+	var transactionhist TransactionDetail
+	if er := json.Unmarshal([]byte(transactionhistoryJSON), &transactionhist); er != nil {
+		t.Error("Okx Transaction Detail error", er.Error())
+	}
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, er := ok.GetTransactionDetailsLast3Days(context.Background(), &TransactionDetailRequestParams{
+		InstrumentType: "MARGIN",
+	}); er != nil {
+		t.Error("Okx GetTransactionDetailsLast3Days() error", er)
+	}
+	if _, er := ok.GetTransactionDetailsLast3Months(context.Background(), &TransactionDetailRequestParams{
+		InstrumentType: "MARGIN",
+	}); er != nil {
+		t.Error("Okx GetTransactionDetailsLast3Days() error", er)
+	}
+}
+
+func TestStopOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.SkipNow()
+	}
+	if _, er := ok.PlaceStopOrder(context.Background(), &AlgoOrderParams{
+		TakeProfitTriggerPriceType: "index",
+
+		InstrumentID: "BTC-USDT",
+		OrderType:    "move_order_stop",
+		Side:         order.Buy,
+	}); er != nil && errors.Is(er, errMissingTakeProfitTriggerPrice) {
+		t.Errorf("Okx StopOrderParams() expecting %v, but found %v", errMissingTakeProfitTriggerPrice, er)
+	}
+	if _, er := ok.PlaceStopOrder(context.Background(), &AlgoOrderParams{
+		TakeProfitTriggerPriceType: "index",
+		InstrumentID:               "BTC-USDT",
+		OrderType:                  "move_order_stop",
+		Side:                       order.Buy,
+	}); er != nil && errors.Is(er, errMissingTakeProfitTriggerPrice) {
+		t.Errorf("Okx StopOrderParams() expecting %v, but found %v", errMissingTakeProfitTriggerPrice, er)
+	}
+	if _, er := ok.PlaceTrailingStopOrder(context.Background(), &AlgoOrderParams{
+		CallbackRatio: 0.01,
+		InstrumentID:  "BTC-USDT",
+		OrderType:     "move_order_stop",
+		Side:          order.Buy,
+	}); er != nil {
+		t.Error("Okx PlaceTrailingStopOrder error", er)
+	}
+	if _, er := ok.PlaceIcebergOrder(context.Background(), &AlgoOrderParams{
+		PriceLimit:  100.22,
+		SizeLimit:   9999.9,
+		PriceSpread: "0.04",
+
+		InstrumentID: "BTC-USDT",
+		OrderType:    "move_order_stop",
+		Side:         order.Buy,
+	}); er != nil {
+		t.Error("Okx PlaceIceburgOrder() error", er)
+	}
+	if _, er := ok.PlaceTWAPOrder(context.Background(), &AlgoOrderParams{
+		PriceLimit:   100.22,
+		SizeLimit:    9999.9,
+		PriceSpread:  "0.4",
+		TimeInterval: kline.ThreeDay,
+	}); er != nil {
+		t.Error("Okx PlaceTWAPOrder() error", er)
+	}
+	if _, er := ok.TriggerAlgoOrder(context.Background(), &AlgoOrderParams{
+		TriggerPriceType: "mark",
+
+		InstrumentID: "BTC-USDT",
+		OrderType:    "move_order_stop",
+		Side:         order.Buy,
+	}); er != nil {
+		t.Error("Okx TriggerAlogOrder() error", er)
+	}
```

Comment:
Please have a dedicated test for these functions with more than just first error responses. I'd like to see a request be made